Tools/AP_Periph/networking.cpp

```diff
@@ -0,0 +1,223 @@
+    // @Group: P5_
+    // @Path: ../libraries/AP_Networking/AP_Networking_port.cpp
+    AP_SUBGROUPINFO(networking_lib.ports[4], "P5_", 6, Networking_Periph, AP_Networking::Port),
+#endif
+
+#if AP_NETWORKING_NUM_PORTS > 5
+    // @Group: P6_
+    // @Path: ../libraries/AP_Networking/AP_Networking_port.cpp
+    AP_SUBGROUPINFO(networking_lib.ports[5], "P6_", 7, Networking_Periph, AP_Networking::Port),
+#endif
+
+#if AP_NETWORKING_NUM_PORTS > 6
+    // @Group: P7_
+    // @Path: ../libraries/AP_Networking/AP_Networking_port.cpp
+    AP_SUBGROUPINFO(networking_lib.ports[6], "P7_", 8, Networking_Periph, AP_Networking::Port),
+#endif
+
+#if AP_NETWORKING_NUM_PORTS > 7
+    // @Group: P8_
+    // @Path: ../libraries/AP_Networking/AP_Networking_port.cpp
+    AP_SUBGROUPINFO(networking_lib.ports[7], "P8_", 9, Networking_Periph, AP_Networking::Port),
+#endif
+
+#if AP_NETWORKING_NUM_PORTS > 8
+    // @Group: P9_
+    // @Path: ../libraries/AP_Networking/AP_Networking_port.cpp
+    AP_SUBGROUPINFO(networking_lib.ports[8], "P9_", 10, Networking_Periph, AP_Networking::Port),
+#endif
+
+
+
+#if HAL_PERIPH_NETWORK_NUM_PASSTHRU > 0
+    // @Group: PASS1_
+    // @Path: networking.cpp
+    AP_SUBGROUPINFO(passthru[0], "PASS1_", 11, Networking_Periph, Passthru),
+#endif
+
+#if HAL_PERIPH_NETWORK_NUM_PASSTHRU > 1
+    // @Group: PASS2_
+    // @Path: networking.cpp
+    AP_SUBGROUPINFO(passthru[1], "PASS2_", 12, Networking_Periph, Passthru),
+#endif
+
+#if HAL_PERIPH_NETWORK_NUM_PASSTHRU > 2
+    // @Group: PASS3_
+    // @Path: networking.cpp
+    AP_SUBGROUPINFO(passthru[2], "PASS3_", 13, Networking_Periph, Passthru),
+#endif
+
+#if HAL_PERIPH_NETWORK_NUM_PASSTHRU > 3
+    // @Group: PASS4_
+    // @Path: networking.cpp
+    AP_SUBGROUPINFO(passthru[3], "PASS4_", 14, Networking_Periph, Passthru),
+#endif
+
+#if HAL_PERIPH_NETWORK_NUM_PASSTHRU > 4
+    // @Group: PASS5_
+    // @Path: networking.cpp
+    AP_SUBGROUPINFO(passthru[4], "PASS5_", 15, Networking_Periph, Passthru),
+#endif
+
+#if HAL_PERIPH_NETWORK_NUM_PASSTHRU > 5
+    // @Group: PASS6_
+    // @Path: networking.cpp
+    AP_SUBGROUPINFO(passthru[5], "PASS6_", 16, Networking_Periph, Passthru),
+#endif
+
+#if HAL_PERIPH_NETWORK_NUM_PASSTHRU > 6
+    // @Group: PASS7_
+    // @Path: networking.cpp
+    AP_SUBGROUPINFO(passthru[6], "PASS7_", 17, Networking_Periph, Passthru),
+#endif
+
+#if HAL_PERIPH_NETWORK_NUM_PASSTHRU > 7
+    // @Group: PASS8_
+    // @Path: networking.cpp
+    AP_SUBGROUPINFO(passthru[7], "PASS8_", 18, Networking_Periph, Passthru),
+#endif
+
+#if HAL_PERIPH_NETWORK_NUM_PASSTHRU > 8
+    // @Group: PASS9_
+    // @Path: networking.cpp
+    AP_SUBGROUPINFO(passthru[8], "PASS9_", 19, Networking_Periph, Passthru),
+#endif
+
+    AP_GROUPEND
+};
+
+
+const AP_Param::GroupInfo Networking_Periph::Passthru::var_info[] = {
+    AP_GROUPINFO_FLAGS("ENABLE", 1,  Networking_Periph::Passthru, enabled, 0, AP_PARAM_FLAG_ENABLE),
+    AP_GROUPINFO("EP1", 2,  Networking_Periph::Passthru, ep1, -1),
```

Comment:
needs param doc markup

Reply:
param docs added! I'll see what CI says about if I need them in a separate file or not